
bwa.sh	Basic bwa alignment
genomeStats.sh	makes STAR index which provides fasta length metrics
htseq_count.sh	pipeline to trim, map 3prime RNAseq read data and combine to read table.
jellyfish.sh	old way of kmer genome size estimation
kmc.sh	new way of kmer genome size estimation, uses space rather than ram
ngmlr.sh	aligner for SV with sniffles
nucmer.sh	mummer SV identification from reads to genome mapping
oneLiners.sh	bash common  bioinformatics manipulations
pacbioSV.sh	approach to identify SV from pacbio mapping to assembly
pblat.sh	align transcriptome to genome to manually check SV
ragtag.sh	reads to genome for assembly correction and SV identification
shorts.sh	Small scripts while still in production.
supernova.sh	assemble 10x data
trimmomatic.sh	basic script to trim reads
seq_extractor.py	script that extracts sequences from fasta based on names in list.